apollo-compiler/src/test/graphql/com/example/fragment_friends_connection/FragmentFriendsConnectionExpected.java

```diff
@@ -1,15 +1,56 @@
 package com.example.fragment_friends_connection;
 
+import com.apollostack.api.GraphQLQuery;
+import java.lang.Override;
+import java.lang.String;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 import javax.annotation.Nullable;
 
-public interface FragmentFriendsConnection {
-  @Nullable Hero hero();
+public final class FragmentFriendsConnection implements GraphQLQuery<FragmentFriendsConnection.Data> {
+  private final String OPERATION_DEFINITION = "query FragmentFriendsConnection {\n"
+      + "  hero {\n"
+      + "    __typename\n"
+      + "    ...HeroDetails\n"
+      + "  }\n"
+      + "}";
 
-  interface Hero {
-    Fragments fragments();
+  private final List<String> FRAGMENT_DEFINITIONS = Collections.unmodifiableList(Arrays.asList(
+    "fragment HeroDetails on Character {\n"
+        + "  __typename\n"
+        + "  name\n"
+        + "  friendsConnection {\n"
+        + "    totalCount\n"
+        + "    edges {\n"
+        + "      node {\n"
+        + "        __typename\n"
+        + "        name\n"
+        + "      }\n"
+        + "    }\n"
+        + "  }\n"
+        + "}"
+  ));
 
-    interface Fragments {
-      HeroDetails heroDetails();
+  @Override
```

Comment:
oh, wait override belongs to method not to return type. So JavaPoet allows only to add annotation to method that will lead to what we have now:
```
@Override
  public String getOperationDefinition() {
    return OPERATION_DEFINITION;
  }
```
We can annotate return type but it will generate:
```
public @Override String getOperationDefinition() {
    return OPERATION_DEFINITION;
  }
```
That is not right as override in this case belongs to type but not to method itself